### PR TITLE
Kate drafting fact sheets

### PR DESCRIPTION
**Pull request by katherineathom**
Let me know if you see anything weird. Didn't fully clean up or comment but... Let me know if you have any questions.

### CODE REVIEW

File: simple-factsheet.Rmd [hunk 1]
```diff
@@ -23,91 +28,401 @@ library(tidyverse)
...
+state_text <- ifelse(params$state == "District of Columbia",
+                     paste("the",params$state,sep=" "),
+                     params$state)
+
+states_subset <- read_excel(filepath) %>%
+  filter(StateName == params$state) %>%
+  rename_at(vars(contains("1718")),~sub("_1718", "", .)) %>%
+  
+  # prepare overall cOI population number - rounded thousand or numeric + "million" text
+  mutate(text_1_popcount = case_when(total_popcount_coi < 1000000 ~ formatC(total_popcount_coi,
```
Comment (awunderground) [inline simple-factsheet.Rmd]: Use `if_else()` instead of `case_when()` when there is only one condition.

File: simple-factsheet.Rmd [hunk 2]
```diff
@@ -23,91 +28,401 @@ library(tidyverse)
...
+                                                                            big.mark =","),
+                                     TRUE ~ paste(round((total_popcount_coi/1000000),1), "million"))) %>%
+  
+  # convert decimals to be communicated as percentages - no decimals except under 1% 
+  # Except for those that will only be used in data vis 
+  mutate_at(vars(starts_with("pct"),
+                 starts_with("share")), topercent) %>%
+  
+  # prepare strings that'll change
+  mutate(text_2_citizens = case_when(share_uscitizen_coi >= 85 ~ "Almost all",
```
Comment (awunderground) [inline simple-factsheet.Rmd]: I recommend aligning the `~` in `case_when()`.

File: simple-factsheet.Rmd [hunk 3]
```diff
@@ -23,91 +28,401 @@ library(tidyverse)
...
+                                     share_uscitizen_coi < 85 & share_uscitizen_coi >= 50 ~ "Most",
+                                     share_uscitizen_coi < 50 ~ "Many",
+                                     TRUE ~ NA_character_),
```
Comment (awunderground) [inline simple-factsheet.Rmd]: Don't add `TRUE` to `case_when()` to add `NA`. The `NA` will implicitly be added to all observations that don't meet the conditions.

File: simple-factsheet.Rmd [hunk 4]
```diff
@@ -23,91 +28,401 @@ library(tidyverse)
...
+         
+         text_3_citizens = case_when(share_prtnatcit_coi >= 50 ~ "most",
+                                     share_prtnatcit_coi < 50 ~ "many",
+                                     TRUE ~ NA_character_),
+         
+         temp_working_diff = share_workingfam_coi - share_workingfam_nat,
+         
+         text_4_employment = case_when(abs(temp_working_diff) < 3 ~ "similar for children of immigrants and",
```
Comment (awunderground) [inline simple-factsheet.Rmd]: `case_when()` evaluates the logical conditions in order from top to bottom. The following is a clearer way to write this section:

```
case_when(temp_working_diff <= -6     ~ "lower for children of immigrants than",
                  temp_working_diff <= -3     ~ "slightly lower for children of immigrants than",
                  temp_working_diff < 3 ~ "similar for children of immigrants and",
                  temp_working_diff < 6         ~ "slightly higher for children of immigrants than",
                  TRUE                                     ~ "higher for children of immigrants than")
```

File: simple-factsheet.Rmd [hunk 5]
```diff
@@ -23,91 +28,401 @@ library(tidyverse)
...
+         
+         text_6_race = case_when((share_asian_coi >= 75 |
+                                  share_black_coi >= 75 | 
+                                  share_hispanic_coi >= 75 |
+                                  share_white_coi >= 75 ~ " fairly "),
+                                 TRUE ~ " "),
+         
+         temp_1_uninsure = (share_noinsure_all_0809) - (share_noinsure_all_1617),
+         
+         text_1_uninsure = case_when(temp_1_uninsure >= 1 ~ "fallen",
```
Comment (awunderground) [inline simple-factsheet.Rmd]: Conditions 1 and 2 include all real numbers so I don't think condition 3 (`temp_1_uninsure <= 1 ~ "risen"`)` will ever be reached.

File: simple-factsheet.Rmd [hunk 6]
```diff
@@ -23,91 +28,401 @@ library(tidyverse)
...
+                                     abs(temp_1_uninsure) < 1 ~ "not changed dramatically",
+                                     temp_1_uninsure <= 1 ~ "risen",
+                                     TRUE ~ NA_character_)) %>%
+  select(-starts_with("temp"))
+  
+# Dataset for race paragraph 
+# handling oth/mult race separately since it doesn't move around like the other racial/ethnic groupings
+race_subset <- states_subset %>%
+ select(c("StateName","share_asian_coi","share_black_coi","share_hispanic_coi","share_white_coi",
```
Comment (awunderground) [inline simple-factsheet.Rmd]: Why are you including quotes and `c()` here?

File: simple-factsheet.Rmd [hunk 7]
```diff
@@ -23,91 +28,401 @@ library(tidyverse)
...
+  pivot_longer(cols = starts_with("share"),
+              names_to = "race",
+              values_to = "share") %>%
+  mutate(race = case_when(str_detect(race,"asian") ~ "Asian",
+                          str_detect(race,"black") ~ "Black",
+                          str_detect(race,"hispanic") ~ "Hispanic",
+                          str_detect(race,"white") ~ "white")) %>%
+  rename(fastest = race_COIpop_growmost_0506to1718) %>%
+  mutate(fastest = ifelse(tolower(race) == fastest,1,0)) %>%
+  arrange(-share) %>%
```
Comment (awunderground) [inline simple-factsheet.Rmd]: This can be rewritten with something like `dense_rank()` from `library(dplyr)`.

File: simple-factsheet.Rmd [hunk 8]
```diff
@@ -23,91 +28,401 @@ library(tidyverse)
...
+  urban_blue <- urbnthemes::palette_urbn_diverging[7]
+  urban_black <- urbnthemes::palette_urbn$categorical[[3]][3]
+  urban_grey <- urbnthemes::palette_urbn$categorical[[4]][4]
+  urban_lightblue <- urbnthemes::palette_urbn$sequential[[2]][1]
+  
+## Creating Figure 1
+  
+  fig1_data <- states_subset %>%
+    select(c("share_natpar","share_uscitz_coi","share_noncitz_coi","share_unkpar")) %>%
+    pivot_longer(everything()) %>%
```
Comment (awunderground) [inline simple-factsheet.Rmd]: I don't think `everything()` is necessary.

File: simple-factsheet.Rmd [hunk 9]
```diff
@@ -23,91 +28,401 @@ library(tidyverse)
...
+
+  urban_blue <- urbnthemes::palette_urbn_diverging[7]
+  urban_black <- urbnthemes::palette_urbn$categorical[[3]][3]
+  urban_grey <- urbnthemes::palette_urbn$categorical[[4]][4]
+  urban_lightblue <- urbnthemes::palette_urbn$sequential[[2]][1]
+  
+## Creating Figure 1
+  
+  fig1_data <- states_subset %>%
+    select(c("share_natpar","share_uscitz_coi","share_noncitz_coi","share_unkpar")) %>%
```
Comment (awunderground) [inline simple-factsheet.Rmd]: Select should work without `c()` and `""`

File: simple-factsheet.Rmd [hunk 10]
```diff
@@ -23,91 +28,401 @@ library(tidyverse)
...
+    pivot_longer(everything()) %>%
+    rename(par_status = name,
+           share = value) %>%
+    mutate(par_status = factor(par_status),
+           par_status = fct_relevel(par_status,
+                                    c("share_unkpar",
+                                      "share_noncitz_coi",
+                                      "share_uscitz_coi",
+                                      "share_natpar")), 
+           var = rep("citizenship",times=nrow(.)),
```
Comment (awunderground) [inline simple-factsheet.Rmd]: I think `var = "citizenship"` will work because `mutate()` is vectorized.

File: simple-factsheet.Rmd (hunk 8 above)
Comment (katherineathom) [reply simple-factsheet.Rmd]: doesn't run for me without `everything()`! I get `Error: `cols` must select at least one column.`

File: simple-factsheet.Rmd (hunk 4 above)
Comment (katherineathom) [reply simple-factsheet.Rmd]: got it; thank you.

File: simple-factsheet.Rmd (hunk 5 above)
Comment (katherineathom) [reply simple-factsheet.Rmd]: `         text_1_uninsure = case_when(temp_1_uninsure >= 1 ~ "fallen"
                                     temp_1_uninsure > -1 ~ "not changed dramatically",
                                     TRUE ~ "risen" `
-- confirming this makes sense?

Comment (katherineathom) [reply simple-factsheet.Rmd]: resolved, I think

**Pull request closed without merging**